File: software/examples/go/example_alarm.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/piezo_speaker_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Piezo Speaker Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	ps, _ := piezo_speaker_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// 10 seconds of loud annoying fast alarm
	ps.SetAlarm(800, 2000, 10, 1, 10, 10000)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}

File: software/examples/c/example_alarm.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_piezo_speaker_v2.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change XYZ to the UID of your Piezo Speaker Bricklet 2.0

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	PiezoSpeakerV2 ps;
	piezo_speaker_v2_create(&ps, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// 10 seconds of loud annoying fast alarm
	piezo_speaker_v2_set_alarm(&ps, 800, 2000, 10, 1, 10, 10000);

	printf("Press key to exit\n");
	getchar();
	piezo_speaker_v2_destroy(&ps);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}
